### PR TITLE
KOGITO-5508 InMemory Data Index using quarkus http instead of Kafka

### PR DESCRIPTION
**Pull request by cristianonicolai**

### CODE REVIEW

**Comment by cristianonicolai** [comment]
Jenkins run native

File: apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-quarkus/src/test/java/org/kie/kogito/index/inmemory/ProcessDataIndexInMemoryIT.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.kie.kogito.index.quarkus.InMemoryTestProfile;
+
+import io.quarkus.test.junit.QuarkusTest;
+import io.quarkus.test.junit.TestProfile;
+
+@QuarkusTest
+@TestProfile(InMemoryTestProfile.class)
+public class ProcessDataIndexInMemoryIT extends AbstractProcessDataIndexIT {
+
+    @ConfigProperty(name = DataIndexPostgreSqlQuarkusTestResource.KOGITO_DATA_INDEX_SERVICE_URL)
```
Comment (nmirasch) [inline apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-quarkus/src/test/java/org/kie/kogito/index/inmemory/ProcessDataIndexInMemoryIT.java]: @cristianonicolai should this reference be to  DataIndexInMemoryQuarkusTestResource.DATA_INDEX_SERVICE_URL instead?
(Silly thing regarding we are using the same value "kogito.dataindex.http.url", just a matter of consistence)

**Review by nmirasch — APPROVED** [review]
I just added a minor comment, but overall the code looks good to me and works fine.

File: apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-quarkus/src/test/java/org/kie/kogito/index/inmemory/ProcessDataIndexInMemoryIT.java (hunk 1 above)
Comment (cristianonicolai) [reply apps-integration-tests/integration-tests-data-index-service/integration-tests-data-index-service-quarkus/src/test/java/org/kie/kogito/index/inmemory/ProcessDataIndexInMemoryIT.java]: oooops :)

**Comment by danielezonca** [comment]
@cristianonicolai 
Have you considered to use reactive stream in-memory connector to replace Kafka when running in devmode?
https://quarkus.io/guides/kafka#testing-a-kafka-application

**Comment by cristianonicolai** [comment]
> @cristianonicolai Have you considered to use reactive stream in-memory connector to replace Kafka when running in devmode? https://quarkus.io/guides/kafka#testing-a-kafka-application

yep, but in order to connect the image of the data index service in dev mode, to the runtime, quarkus-http actually fits perfectly

**Comment by danielezonca** [comment]
> > @cristianonicolai Have you considered to use reactive stream in-memory connector to replace Kafka when running in devmode? https://quarkus.io/guides/kafka#testing-a-kafka-application
> 
> yep, but in order to connect the image of the data index service in dev mode, to the runtime, quarkus-http actually fits perfectly

Ok my bad I was thinking at a scenario where data-index "core" is used as library/dependency inside the dev extension (so data-index API are exposed directly by the runtime thanks to this dev extension) and not as docker image.
Thanks for the clarification 👍

**Review by radtriste — APPROVED** [review]
lgtm

**Comment by radtriste** [comment]
jenkins run native

**Comment by radtriste** [comment]
@cristianonicolai looks like the `NativeProcessDataIndex**IT` tests are failing. Can you have a look ?

**Comment by cristianonicolai** [comment]
@radtriste @nmirasch it seems a problem with runtimes, I've created a follow up JIRA https://issues.redhat.com/browse/KOGITO-6116 and will leave native tests disabled for now.

**Comment by cristianonicolai** [comment]
@tiagodolphine @ricardozanini ready to merge

**Pull request merged**